### ISSUE TITLE
Liquid: liquid markup in style or script tags

### ISSUE THREAD

**Issue by macdonaldr93** [prettydiff/prettydiff#285]
If `liquid` markup is within a `script` or `style` tag, the beautifier tries to format it to the respective language.

## Source
```
{% if exists %}
  <script>
    (function($) {
      'use strict';

      $(document).ready(function(){
        var x = {{ settings.x  }};
        var y = {{ settings.y | json }};

        {% assign z_array = settings.z | strip_newlines | split: ',' %}
        var zArray = [
          {% for a in z %}
          {% assign a_x = a | split: ':' | first %}
          {% assign a_y = a | split: ':' | last | plus: 0 %}
          { 
            x:'{{ a_x }}', 
            y: {{ a_y }}
          }{% unless forloop.last %},{% endunless %}
          {% endfor %}
        ];
     });
    })(jQuery);
  </script>
{% endif %}
```

## Output
```
{% if exists %}
  <script>
    (function ($) {
      'use strict';

      $(document).ready(function () {
        var x = {
          {
            settings.x
          }
        };
        var y = {
          {
            settings.y | json
          }
        };

        { % assign z_array = settings.z | strip_newlines | split : ',' %
        }
        var zArray = [
          { % for a in z %
          }
          { % assign a_x = a | split: ':' | first %
          }
          { % assign a_y = a | split: ':' | last | plus : 0 %
          }
          {
            x: '{{ a_x }}',
            y: {
              {
                a_y
              }
            }
          }
          { % unless forloop.last %
          }, { % endunless %
          }
          { % endfor %
          }
        ];
      });
    })(jQuery);
  </script>
{% endif %}
```

**Comment by prettydiff** [comment]
Try it out at http://prettydiff.com/ignore/testlocation/index.xhtml?m=beautify

**Comment by macdonaldr93** [comment]
Definitely a lot better!

## Output

```javascript
{% if exists %}
    <script>
        (function ($) {
            'use strict';

            $(document).ready(function () {
                var x = {
                    {
                        settings.x
                    }
                };
                var y = {
                    {
                        settings.y | json
                    }
                };

                {% assign z_array = settings.z | strip_newlines | split: ',' %}
                var zArray = [
                    {% for a in z %}
                    {% assign a_x = a | split: ':' | first %}
                    {% assign a_y = a | split: ':' | last | plus: 0 %}
                    {
                        x: '{{ a_x }}',
                        y: {
                            {
                                a_y
                            }
                        }
                    }
                    {% unless forloop.last %},
                    {% endunless %}
                    {% endfor %}
                ];
            });
        })(jQuery);
    </script>
{% endif %}
```

## Issues
There are still a few issues in here:

### Example 1
```javascript
var x = {
    {
        settings.x
    }
};
var y = {
    {
        settings.y | json
    }
};
```

should be:

```javascript
var x = {{ settings.x  }};
var y = {{ settings.y | json }};
```

### Example 2
and this is related to the same syntax and one indent missing:

```javascript
var zArray = [
    {% for a in z %}
    {% assign a_x = a | split: ':' | first %}
    {% assign a_y = a | split: ':' | last | plus: 0 %}
    {
        x: '{{ a_x }}',
        y: {
            {
                a_y
            }
        }
    }
    {% unless forloop.last %},
    {% endunless %}
    {% endfor %}
];
```

should be:

```javascript
var zArray = [
  {% for a in z %}
    {% assign a_x = a | split: ':' | first %}
    {% assign a_y = a | split: ':' | last | plus: 0 %}
    { 
      x:'{{ a_x }}', 
      y: {{ a_y }}
    }{% unless forloop.last %},{% endunless %}
  {% endfor %}
];
```

**Comment by prettydiff** [comment]
I see from https://github.com/Shopify/liquid/wiki/Liquid-for-Designers that this is the tag list:

* assign - Assigns some value to a variable
* capture - Block tag that captures text into a variable
* case - Block tag, its the standard case...when block
* comment - Block tag, comments out the text in the block
* cycle - Cycle is usually used within a loop to alternate between values, like colors or DOM classes.
* for - For loop
* break - Exits a for loop
* continue Skips the remaining code in the current for loop and continues with the next loop
* if - Standard if/else block
* include - Includes another template; useful for partials
* raw - temporarily disable tag processing to avoid syntax conflicts.
* unless - Mirror of if statement

Which of those do not get an end tag?

**Comment by prettydiff** [comment]
In addition to the supplied code I have also been running tests with this code:

    {% if exists %}
        <style>
            a {
                background: {{ settings.x  }};
                color     : {{ settings.y | json }};

                {% assign z_array = settings.z | strip_newlines | split: ',' %}
                b {
                    {% for a in z %}
                        {% assign a_x = a | split: ':' | first %}
                        {% assign a_y = a | split: ':' | last | plus: 0 %}
                        c {
                            x: {{ a_x }};
                            y: {{ a_y }};
                        }
                        {% unless forloop.last %},{% endunless %}
                    {% endfor %}
                }
            }

        </style>
    {% endif %}

Updated code, please review: http://prettydiff.com/ignore/testlocation/index.xhtml?m=beautify

**Comment by macdonaldr93** [comment]
I'm getting you the info required around the different `liquid` tags. I should have it back to you shortly.

**Comment by macdonaldr93** [comment]
Okay so here we go:

This is a pretty solid resource to reference: [Liquid Cheatsheet](http://cheat.markdunkley.com/).

Also, any of the tags below have additional attributes and/or filters. This is simply to identify the tag.

## Tags with end

- Comment

  ```
  {% comment %}
  {% endcomment %}
  ```

- Raw

  ```
  {% raw %}
  {% endraw %}
  ```

- If

  ```
  {% if %}
  {% elsif %}
  {% else %}
  {% endif %}
  ```

- Unless

  ```
  {% unless %}
  {% endunless %}
  ```

- Case

  ```
  {% case %} 
  {% when %}
  {% else %}
  {% endcase %}
  ```

- For

  ```
  {% for %} 
  {% endfor %}
  ```

- Tablerow

  ```
  {% tablerow %}
  {% endtablerow %}
  ```

- Capture

  ```
  {% capture %}
  {% endcapture %}
  ```

- Form

  ```
  {% form %}
  {% endform %}
  ```

- Paginate

  ```
  {% paginate %}
  {% endpaginate %}
  ```

## Tags without end

- Cycle
`{% cycle %}`

- Assign
`{% assign %}`

- Increment
`{% increment %}`

- Decrement
`{% decrement %}`

- Include
`{% include %}`

- Layout
`{% layout %}`

**Comment by prettydiff** [comment]
Released with version v1.16.28.
